packages/terra-date-time-picker/src/DateTimePicker.jsx

```diff
@@ -0,0 +1,368 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames/bind';
+import 'terra-base/lib/baseStyles';
+import DatePicker from 'terra-date-picker';
+import TimeInput from 'terra-time-input';
+import DateUtil from 'terra-date-picker/lib/DateUtil';
+import styles from './DateTimePicker.scss';
+import DateTimeUtils from './DateTimeUtils';
+import TimeClarification from './_TimeClarification';
+
+const cx = classNames.bind(styles);
+
+const propTypes = {
+  /**
+   * Custom input attributes to apply to the date input. Use the name prop to set the name for the date input.
+   * Do not set the name in inputAttribute as it will be ignored.
+   */
+  dateInputAttributes: PropTypes.object,
+  /**
+   * An array of ISO 8601 string representation of the dates to disable in the picker.
+   */
+  excludeDates: PropTypes.arrayOf(PropTypes.string),
+  /**
+   * A function that gets called for each date in the picker to evaluate which date should be disabled.
+   * A return value of true will be enabled and false will be disabled.
+   */
+  filterDate: PropTypes.func,
+  /**
+   * An array of ISO 8601 string representation of the dates to enable in the picker.
+   * All Other dates will be disabled.
+   */
+  includeDates: PropTypes.arrayOf(PropTypes.string),
+  /**
+   * An ISO 8601 string representation of the maximum date time.
+   */
+  maxDateTime: PropTypes.string,
+  /**
+   * An ISO 8601 string representation of the minimum date time.
+   */
+  minDateTime: PropTypes.string,
+  /**
+   * Name of the date input. The name should be unique.
+   */
+  name: PropTypes.string.isRequired,
+  /**
+   * A callback function to execute when a valid date is selected or entered.
+   * The first parameter is the event. The second parameter is the changed input value.
+   */
+  onChange: PropTypes.func,
+  /**
+   * A callback function to execute when a change is made in the date or time input.
+   * The first parameter is the event. The second parameter is the changed input value.
+   */
+  onChangeRaw: PropTypes.func,
+  /**
+   * A callback function to let the containing component (e.g. modal) to regain focus.
+   */
+  releaseFocus: PropTypes.func,
+  /**
+   * A callback function to request focus from the containing component (e.g. modal).
+   */
+  requestFocus: PropTypes.func,
+  /**
+   * Custom input attributes to apply to the time input. Use the name prop to set the name for the time input.
+   * Do not set the name in inputAttribute as it will be ignored.
+   */
+  timeInputAttributes: PropTypes.object,
+  /**
+   * An ISO 8601 string representation of the initial value to show in the date and time inputs.
+   */
+  value: PropTypes.string,
+};
+
+const defaultProps = {
+  dateInputAttributes: undefined,
+  excludeDates: undefined,
+  filterDate: undefined,
+  includeDates: undefined,
+  maxDateTime: undefined,
+  minDateTime: undefined,
+  name: undefined,
```

Comment:
do you really need to default these to undefined?  wouldn't they be undefined automatically if you didn't default them to undefined

Reply:
At one point the value and defaultValue were defaulted to null, which causes a warning, and I had to default to undefined to make it work as desired.